### ISSUE TITLE
OBS takes 30s to show picker (and does not remember picked window accross reboots)

### ISSUE THREAD

**Issue by TheSunCat** [hyprwm/xdg-desktop-portal-hyprland#68]
When I start OBS, the screen capture shows black. I have to wait 30s before clicking the "Select monitor" button does anything. If I clicked it X times during these initial 30s, X picker windows open. This is reproducible every time I reboot, as OBS forgets the picked screen/window and I have to pick it again (waiting the 30s). This does not happen screen sharing with WebCord or Firefox.

**Comment by TheSunCat** [comment]
Possibly relevant logs from `systemctl --user status xdg-desktop-portal`:
```
Jul 13 18:27:58 apg xdg-desktop-por[2391]: A backend call failed: Remote peer disconnected
Jul 13 18:27:58 apg xdg-desktop-por[2391]: Error deleting permission: No entry for fbd03552-8a68-4670-906b-5cbe101de458
Jul 13 18:28:23 apg xdg-desktop-por[2391]: Failed to close session implementation: Timeout was reached
```

**Comment by vaxerski** [comment]
dbus wonkery on your end

**Comment by Mikilio** [comment]
I have the same issue and I'm basically using the default NixOS/home-manager setup. To shed some light I have used dbus-monitor to record dbus activity during the bug:
[Gist](https://gist.github.com/Mikilio/f40089df9f9ae03b60223464f004a04e)

**Comment by aimixess** [comment]
Hello,

I also encounter this issue on Arch with OBS 29.0.2-5.
```Aug 13 22:36:03 oufzor systemd[1387]: Started Portal service.
Aug 13 22:36:17 oufzor xdg-desktop-por[1531]: A backend call failed: Message recipient disconnected from message bus without replying
Aug 13 22:36:17 oufzor xdg-desktop-por[1531]: Error deleting permission: No entry for bf62b8c1-8d21-49d6-b014-ebf23958d4d2
```

**Comment by ymnejmi** [comment]
I have been facing this problem since I started using hyprland/obs from before march iirc, I changed repos many times (a few arch based and now im on nixos), and it still persists across all of them.
Logs are the same as mentioned above:
`Sep 14 16:32:51 nixos .xdg-desktop-po[1378]: A backend call failed: Message recipient disconnected from message bus without replying
Sep 14 16:32:51 nixos .xdg-desktop-po[1378]: Error deleting permission: No entry for cee3dec4-7f27-461f-908a-4de334aa6066
Sep 14 16:33:16 nixos .xdg-desktop-po[1378]: Failed to close session implementation: Timeout was reached`
I do not know what could make my dbus wonkery, I dont even really fully understand how this portal stuff so I just follow guides and/or random posts on the web, but the only reference I could find to this specific problem is this issue, so I dont even know how to begin to fix it.
I have an nvidia gpu laptop if that is of any relevance.
I would appreciate if anyone who found a solution to share it with us, it is not breaking function but it is annoying to have to select monitor and then to wait for the picker after every reboot.
